ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_device_operation.cpp

```diff
@@ -145,19 +145,17 @@ void BinaryDeviceOperation::validate_on_program_cache_hit(
             batch_size_1_a > batch_size_1_b and batch_size_1_b == 1,
             "ttnn::operations::binary::BinaryDeviceOperation: batch size mismatch");
     }
-    if (height_a != height_b) {
-        TT_ASSERT(
-            height_a > height_b and height_b == 1, "ttnn::operations::binary::BinaryDeviceOperation: height mismatch");
-    }
-    if (width_a != width_b) {
-        TT_ASSERT(
-            width_a > width_b and width_b == 1, "ttnn::operations::binary::BinaryDeviceOperation: width mismatch");
-    }
+
+    TT_ASSERT(height_b == 1, "ttnn::operations::binary::BinaryDeviceOperation: height mismatch");
+    TT_ASSERT(width_b == 1, "ttnn::operations::binary::BinaryDeviceOperation: width mismatch");
```

Comment:
I don't know whats going on here. Do you fatal if height or width is not 1?

Reply:
Yes

Reply:
Fixed the check